src/layer/AbstractSentinelHubV3Layer.ts

```diff
@@ -233,6 +233,16 @@ export class AbstractSentinelHubV3Layer extends AbstractLayer {
       ...this.getFindDatesUTCAdditionalParameters(),
     };
     const response = await axios.post(this.dataset.findDatesUTCUrl, payload);
-    return response.data.map((date: string) => moment.utc(date).toDate());
+    const result = response.data.map((date: string) => moment.utc(date).toDate());
+
+    // S-5P, S-3 and possibly other datasets return the results in reverse order (leastRecent).
```

Comment:
This is also the case with byoc datasets. I'm not sure about sorting. It would probably be more clear and readable, but on the other hand in most of the cases it'll be just sorting already sorted array.

Reply:
This was my assumption too, but @guss84 is right, I should check that this is true... I'll dig up this information and post it here.

Reply:
It seems that this is a bug on the service side (I filed an issue), however it might take some time for this to be fixed. 

I have also noticed that `BYOCLayer` was re-implementing the same logic (but used `await` for `getFindDatesUTCAdditionalParameters`), so I fixed `getFindDatesUTCAdditionalParameters` to be `async` (no harm if function returns an object instead of Promise). 

I have also checked all other layers and added sorting to `WmsLayer`, so now `findDatesUTC` should always return sorted data, no matter which layer it is called on.